src/neomaril_codex/training.py

```diff
@@ -14,6 +17,34 @@
 
 patt = re.compile(r'(\d+)')
 
+class NeomarilTrainingLogger:
+    def __init__(self, name, X_train, y_train):
+        self.name = name
+        self.X_train = X_train
+        self.y_train = y_train
+        self.model_outputs = None
+        self.model = None
+        self.metrics = {}
+        self.params = {}
+        self.requirements = None
```

Comment:
I think we can use [this](https://mlflow.org/docs/latest/python_api/mlflow.models.html#mlflow.models.infer_pip_requirements) to find out if the user dont send the requirements.

Reply:
Solved with a commit in neomaril repo -> https://github.com/datarisk-io/neomaril/pull/311